### PR TITLE
Add JNI bindings for zstd compression of NVCOMP.

### PR DESCRIPTION
**Pull request by firestarman**
LZ4 is already supported, and this PR mainly adds JNI bindings for the ZSTD compression of NVCOMP.

It extracts the common code from LZ4 compressor and decompressor and put it into two new classes `BatchedCompressor` and `BatchedDecompressor`. By extending from these two classes, it adds in the ZSTD support.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: java/src/test/java/ai/rapids/cudf/nvcomp/NvcompTest.java [hunk 1]
```diff
@@ -23,18 +23,29 @@
...
 import java.util.Arrays;
-import java.util.Optional;
 
 public class NvcompTest {
   private static final HostMemoryAllocator hostMemoryAllocator = DefaultHostMemoryAllocator.get();
 
   private static final Logger log = LoggerFactory.getLogger(ColumnVector.class);
 
+  private final long chunkSize = 64 * 1024;
+  private final long targetIntermediteSize = Long.MAX_VALUE;
```
Comment (jlowe) [inline java/src/test/java/ai/rapids/cudf/nvcomp/NvcompTest.java:33]:
```suggestion
  private final long targetIntermediateSize = Long.MAX_VALUE;
```

**Review by jlowe — APPROVED** [review]
Minor typo, lgtm.

File: java/src/test/java/ai/rapids/cudf/nvcomp/NvcompTest.java (hunk 1 above)
Comment (firestarman) [reply java/src/test/java/ai/rapids/cudf/nvcomp/NvcompTest.java:33]: Nice catch. But I'll merge this first, and correct typo in the future PRs.

**Pull request merged**